File: app/src/main/java/com/mapxus/mapxusmapandroiddemo/BaseApplication.java
package com.mapxus.mapxusmapandroiddemo;

import android.app.Application;

import com.mapxus.map.mapxusmap.api.map.MapxusMapContext;

public class BaseApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        MapxusMapContext.init(getApplicationContext());
    }
}

File: app/src/main/java/com/mapxus/mapxusmapandroiddemo/BaseApplication.kt
package com.mapxus.mapxusmapandroiddemo

import android.app.Application
import com.mapxus.map.auth.CognitoContext
import com.mapxus.map.mapxusmap.api.map.MapxusMapContext

class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        CognitoContext.registerWithApiKey(
            applicationContext,
            BuildConfig.MAPXUS_APPID,
            BuildConfig.MAPXUS_SECRET
        )
        MapxusMapContext.initialize(applicationContext)
    }
}
